enhancements/installer/openstack-baremetal-compute.md

```diff
@@ -0,0 +1,156 @@
+floating IPs might represent an increase in complexity and is currently not a
+goal.
+
+## Proposal
+
+This enhancement is less about adding a new logical piece, than it is about
+uncovering unknowns in a configuration that might work out-of-the-box in
+selected environments.
+
+Therefore, the implementation consists of:
+* identifying a reference architecture
+* anticipating probable issues
+* testing and fixing errors
+
+### Characteristics of the reference architecture
+
+* The choice between deploying a VM or BM node is done by selecting an appropriate Nova flavour
+  * which can be done both in install config and passed to Cluster API provider OpenStack
+  * for the time being, the interactive prompt does ask the flavor for the Compute Nodes specifically; the choice must be made editing `install-config.yaml`
+* The deployment request makes the same API call to Nova as with a VM install (there are no explicit calls to Ironic or anything in the Installer required)
+  * so creating baremetal nodes should work transparently from the installer as well as Cluster API provider OpenStack
+* both VMs and BMs consume the same image from glance
+
+### Probable issues
+
+* The Ignition boot system consisting on a second payload downloaded at runtime
+  requires the baremetal machine to have access to Swift early in the boot
+  process.
+
+* When attaching and detaching, the expected timings must be adjusted to
+  reflect the different boot and shutdown latencies of physical machines
+
+* When attaching, the Certificate Signing Request timeout must be adjusted to
+  reflect the different boot and shutdown latencies of physical machines
+
+### User Stories
+
+#### Increased performance
+
+User wants to run the Compute Nodes on baremetal machines for increased
+performance. The Control Plane sits on regular Nova instances to leverage the
+flexibility of virtual machines.
+
+### Risks and Mitigations
+
+TBD
+
+## Design Details
+
+### Test Plan
+
+The enhancement will be tested using virtual baremetal machines.
+
+TODO: alternatives
+	- OVB on RDO?
+        - MOC is planning on adding Ironic
+	- Install (and snapshot?) an Ironic-enabled OSP on PSI (see Infrared-upshift)
+
+
+TODO: references
+        - https://gitlab.cee.redhat.com/jstransk/infrared-upshift-guide
+	- https://docs.openstack.org/project-deploy-guide/tripleo-docs/latest/environments/virtualbmc.html
+        - https://openstack-virtual-baremetal.readthedocs.io/en/latest/introduction.html
+
+### Graduation Criteria
+
+TBD
+
+##### Removing a deprecated feature
+
+TBD
+
+### Upgrade / Downgrade Strategy
+
+TBD
+
+### Version Skew Strategy
+
+TBD
+
+## Implementation History
+
+TBD
+
+## Drawbacks
+
+TBD
+
+## Alternatives
+
+TBD
+
+## Infrastructure Needed
+
+TODO: CI
+
+[openstack-nova]: https://docs.openstack.org/nova "OpenStack Compute (nova)"
+[openstack-ironic]: https://wiki.openstack.org/wiki/Ironic "OpenStack Bare Metal Provisioning Program"
```

Comment:
Should better point to https://docs.openstack.org/ironic for more consistency with nova above, and because anything on OpenStack's wiki is generally very outdated.